be/src/util/json_converter.cpp

```diff
@@ -56,116 +56,118 @@ class SimdJsonConverter {
-            convert(element.value(), {}, builder);
+            convert(element.value(), {}, false, builder);
         }
         builder->close();
         return Status::OK();
     }
 
-    static inline Status convert(SimdJsonNumber num, std::string_view field_name, vpack::Builder* builder) {
+    static inline Status convert(SimdJsonNumber num, std::string_view field_name, bool has_name,
+                                 vpack::Builder* builder) {
         switch (num.get_number_type()) {
         case SimdJsonNumberType::floating_point_number: {
-            if (!field_name.empty()) {
+            if (has_name) {
                 builder->add(toStringRef(field_name), vpack::Value((num.get_double())));
             } else {
                 builder->add(vpack::Value((num.get_double())));
             }
             break;
         }
         case SimdJsonNumberType::signed_integer: {
-            if (!field_name.empty()) {
+            if (has_name) {
                 builder->add(toStringRef(field_name), vpack::Value((num.get_int64())));
             } else {
                 builder->add(vpack::Value((num.get_int64())));
             }
             break;
         }
         case SimdJsonNumberType::unsigned_integer: {
-            if (!field_name.empty()) {
+            if (has_name) {
                 builder->add(toStringRef(field_name), vpack::Value((num.get_uint64())));
             } else {
                 builder->add(vpack::Value((num.get_uint64())));
             }
             break;
         }
         default:
-            __builtin_unreachable();
+            return Status::InternalError(fmt::format("unsupported json number: {}", num.get_number_type()));
         }
         return Status::OK();
     }
 
-    static inline Status convert(std::string_view str, std::string_view field_name, vpack::Builder* builder) {
-        if (!field_name.empty()) {
+    static inline Status convert(std::string_view str, std::string_view field_name, bool has_name,
+                                 vpack::Builder* builder) {
+        if (has_name) {
             builder->add(toStringRef(field_name), vpack::Value(str));
         } else {
             builder->add(vpack::Value(str));
         }
         return Status::OK();
     }
 
-    static inline Status convert(bool value, std::string_view field_name, vpack::Builder* builder) {
-        if (!field_name.empty()) {
+    static inline Status convert(bool value, std::string_view field_name, bool has_name, vpack::Builder* builder) {
+        if (has_name) {
             builder->add(toStringRef(field_name), vpack::Value(value));
         } else {
             builder->add(vpack::Value(value));
         }
         return Status::OK();
     }
 
-    static inline Status convert_null(std::string_view field_name, vpack::Builder* builder) {
-        if (!field_name.empty()) {
+    static inline Status convert_null(std::string_view field_name, bool has_name, vpack::Builder* builder) {
```

Comment:
using `bool has_name=false` may be better.

Reply:
renamed to `is_object`